File: classTypes4/java/src/Iguana.java
public class Iguana extends ReptilBase {
    private String color = "verde"; // Valor por defecto

    // constructor sin parametros
    public Iguana() {
        // valores por defecto para el constructor de la clase padre (ReptilBase/AnimalBase)
        super("Iguana iguana", 1.5, true); 
        System.out.println("se ha creado una Iguana con valores predeterminados (color: " + this.color + ")");
    }
    
    // metodos obligatorios implementados
    @Override
    public void emitirSonido() {
        System.out.println("ssssssss");
    }

    @Override
    public void tipoDeLocomocion() {
        System.out.println("la iguana camina y a veces trepa");
    }
}

File: classTypes4/c#/Iguana.cs
public class Iguana : ReptilBase {
    // valor por defecto
    private string color = "verde"; 

    // constructor sin parametros
    public Iguana() 
        // valores por defecto para el constructor de la clase padre
        : base("iguana iguana", 1.5, true){
        Console.WriteLine("se ha creado una iguana con valores predeterminados (color: " + this.color.ToLower() + ")");
    }
    
    // metodos obligatorios implementados
    public override void emitirsonido() {
        Console.WriteLine("ssssssss");
    }

    public override void tipodelocomocion() {
        Console.WriteLine("la iguana camina y a veces trepa");
    }
}